src/Voron/Data/Tables/NewPageAllocator.cs

```diff
@@ -210,8 +212,6 @@ public unsafe Page AllocateSinglePage(long nearbyPage)
                                 return _llt.ModifyPage(currentSectionStart + i);
                             }
                         }
-                        if (TryMoveNextCyclic(it, startPage) == false)
```

Comment:
As I understand the fix is the addition of above `continue`. Can you please elaborate more on this removal?

Reply:
You never get there, so this is unnecessary.
If you are inside the inner for loop (line 205), you will find the unset bit, and then you'll return an unoccupied page, because the buffer you are passing on isn't `ulong.MaxValue` (contains only set bits).

Reply:
OK